milli/src/update/index_documents/transform.rs

```diff
@@ -610,6 +482,17 @@ fn validate_document_id(document_id: &str) -> Option<&str> {
     })
 }
 
+/// Drops all the value of type `U` in vec, and reuses the allocation to create a `Vec<T>`.
+///
+/// The size and alignment of T and U must match.
+fn drop_and_reuse<U, T>(mut vec: Vec<U>) -> Vec<T> {
+    debug_assert_eq!(std::mem::align_of::<U>(), std::mem::align_of::<T>());
+    debug_assert_eq!(std::mem::size_of::<U>(), std::mem::size_of::<T>());
+    vec.clear();
+    debug_assert!(vec.is_empty());
+    vec.into_iter().map(|_| unreachable!()).collect()
+}
```

Comment:
Ok, that's what I thought would be good to do instead of importing the whole library and you did it, thank you!